MLOps/CreateConfigSecretScript.py

```diff
@@ -0,0 +1,115 @@
+import requests
+from kubernetes import client, config
+import base64
+import json
+import sys
+from getpass import getpass
+
+"""
+This script can be run from the user's local laptop/computer.
+Creates a namespaced secret (containing a kubeconfig) to be used as a KubeDirector connection.
+The secret name is config-@@@USERNAME@@@
+Purpose: to run kubectl commands from within notebook clusters.
+
+"""
+
+def getSession(ip, username, password):
+    data = {"name":username, "password":password}
+    headers  = {'content-type' : 'application/json'}
+    response = requests.post("http://" + ip + ":8080/api/v1/login", json=data, headers=headers)
+    session=response.headers['Location']
+    return session
+
+
+# List all tenant/namespace pairs.
+def getTenantNamespacePairs(ip, username, password):
+    session = getSession(ip, username, password)
+    headers = { 'X-BDS-SESSION' : session}
+    url = "http://" + ip + ":8080/api/v2/tenant"
+    try:
+        response = requests.request("GET", url, headers=headers)
+    except:
+        raise AssertionError("Error getting tenants, please verify that kubectl is set up on your machine.")
+    allTenants = []
+    for tenant in response.json()['_embedded']['tenants']:
+        if (tenant['tenant_type'] == 'k8s'):
+            allTenants.append([tenant['label']['name'], tenant['tenant_enclosed_properties']['namespace']])
+    if (len(allTenants) == 0):
+        raise AssertionError("No k8s tenants found, please verify that kubectl is set up on your machine.")
+    return allTenants
+
+
+# Prompt user to choose a namespace.
+def getNamespace(allTenants):
+    print('\n List of tenants/namespaces:')
+    for i in range(len(allTenants)):
+        print(' [' + str(i) + '] ' + 'Tenant: ' + allTenants[i][0] + ' Namespace: ' + allTenants[i][1])
+    print('\n')
+    tenant_num = int(input("Please enter the number from the list corresponding to your tenant/namespace: "))
```

Comment:
hmm, this is to cover the case for user being in multiple tenants?

Reply:
Hi Kartik, on Friday I believe you mentioned that there was a difference between tenants and namespaces.
So to reduce user confusion, instead of prompting the user to type in a namespace, my workaround was have the user choose from a list all k8s tenants. What do you think?

Reply:
yes, so once they choose a tenant do we do a GET for that tenant to extract namespace?

Reply:
Tenant name is not necessarily the namespace (in most cases but not always)

Reply:
Yes, I am running GET on /api/v2/tenant to retrieve each individual tenant and the specific namespace associated with it (in the method above named `getTenantNamespacePairs`).

Reply:
👍